cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/executor/DatasetAdminService.java

```diff
@@ -78,9 +85,33 @@ public DatasetSpecification create(Id.DatasetInstance datasetInstanceId, Dataset
       }
 
       DatasetSpecification spec = type.configure(datasetInstanceId.getId(), props);
-      DatasetAdmin admin = type.getAdmin(DatasetContext.from(datasetInstanceId.getNamespaceId()), spec);
+      DatasetContext context = DatasetContext.from(datasetInstanceId.getNamespaceId());
+      DatasetAdmin admin = type.getAdmin(context, spec);
       try {
         admin.create();
+
+        // add system metadata for user datasets only
+        if (isUserDataset(datasetInstanceId)) {
+          Dataset dataset = null;
+          try {
+            try {
```

Comment:
We need two try blocks so that we can write metadata even if instantiating dataset throws exception.